docs/adr/0009-create-a-library-for-reading-the-user-consent-cookie.md

```diff
@@ -0,0 +1,113 @@
+networked dependencies into microservices
+* the need for simplicity and for the fewest number of moving parts
+* the preference for not introducing a requirement for microservices to add additional libraries 
+
+## Considered Options
+
+* Option 1: Add the needed logic to play-ui and play-frontend-hmrc
+* Option 2: Add the needed logic to bootstrap-play
+* Option 3: Create a micro-library to contain the logic, with the intention that this would be added to one of the
+above libraries as a transitive dependency
+* Option 4: Create a hosted API to decode and an accompanying client library
+* Option 5: Do nothing
+
+## Decision Outcome
+
+Chosen option 3: because it meets most if not all the above criteria.
+
+### Positive Consequences
+
+* Services can determine acceptance or rejection of non-essential cookies in a single line of code.
+* Changes to the cookie parsing logic can be rolled out relatively easily by #team-plat-ui
+* Consuming services do not rely on any additional networked dependencies
+* No additional libraries would be needed (assuming the micro-library is added to play-frontend-hmrc or bootstrap-play)
+
+### Negative Consequences
+
+* Any version change to the userConsent cookie has the potential to break functionality relying on non-essential cookies
+in consuming services. We may need to look at introducing the ability to have the concept of minor versions or only increment the version for breaking
+changes e.g. removal or change to the meaning of a cookie type.
+
+## Pros and Cons of the Options
+
+### Option 1: Add the needed logic to play-ui and play-frontend-hmrc
+
+* Good, because play-ui and play-frontend-hmrc already has knowledge of tracking consent
+* Bad, because the logic will have to be duplicated across two libraries.
+* Bad, because domain/business logic arguably doesn't live in a frontend component library
+
+### Option 2: Add the needed logic to bootstrap-play
+
+* Good, because bootstrap-play already has some knowledge of cookies
+* Bad, because we would need to introduce knowledge of tracking consent into bootstrap-play. We already have such knowledge
+in play-ui, play-frontend-hmrc and tracking-consent-frontend. See [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter)
+* Bad because bootstrap-play is not owned by #team-plat-ui, and this would hamper our ability to roll out changes
+to the cookie reading logic as it would require approval by another team.
+
+### Option 3: Create a micro-library to contain the logic
+
+* Good, because #team-plat-ui can own the new library and make changes at the same time as changes to
+tracking-consent-frontend
+* Good, because it defers the decision of which library to transitively add it to allowing us to proceed with implementation.
+As a fallback we can advise teams add it on its own.
+* Good, because it's a simple solution and easy to implement and rollout.
```

Comment:
Suggested additional point:
• Good, because we can revisit the decision regarding bootstrap-play if we own the code and this can be included as a dependency by bootstrap if we decide to